### PR TITLE
Add history deep health check into admin api

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
Add history deep health check into admin api

## Why?
wire up admin deep health check with history deep health check

## How did you test it?
unit test

### CODE REVIEW

File: common/dynamicconfig/constants.go [hunk 1]
```diff
@@ -744,6 +744,11 @@ This config is EXPERIMENTAL and may be changed or removed in a later release.`,
 		true,
 		`MaskInternalOrUnknownErrors is whether to replace internal/unknown errors with default error`,
 	)
+	HistoryHostErrorPercentage = NewGlobalFloatSetting(
+		"frontend.historyHostErrorPercentage",
+		50,
+		`HistoryHostErrorPercentage is the percentage to mark history service as unhealthy`,
```
Comment (laniehei) [inline common/dynamicconfig/constants.go:750]: the percentage of host errors?

File: service/frontend/admin_handler.go [hunk 2]
```diff
@@ -240,12 +256,13 @@ func (adh *AdminHandler) Stop() {
 
 func (adh *AdminHandler) DeepHealthCheck(
 	ctx context.Context,
-	request *adminservice.DeepHealthCheckRequest,
+	_ *adminservice.DeepHealthCheckRequest,
```
Comment (laniehei) [inline service/frontend/admin_handler.go:259]: why not remove?

File: service/frontend/health_check.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	hosts := resolver.AvailableMembers()
+	receiveCh := make(chan enumsspb.HealthState, len(hosts))
+	for _, host := range hosts {
+		go func(hostAddress string) {
+			resp, err := h.healthCheckFn(
+				ctx,
+				hostAddress,
+			)
+			if err != nil {
+				h.logger.Warn("Failed to ping deep health check", tag.Error(err), tag.ServerName(string(h.serviceName)))
```
Comment (laniehei) [inline service/frontend/health_check.go:82]: nit 
```suggestion
				h.logger.Warn("failed to ping deep health check", tag.Error(err), tag.ServerName(string(h.serviceName)))
```

File: service/frontend/health_check.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		logger:                logger,
+	}
+}
+
+func (h *healthCheckerImpl) Check(ctx context.Context) (enumsspb.HealthState, error) {
+	resolver, err := h.membershipMonitor.GetResolver(h.serviceName)
+	if err != nil {
+		return enumsspb.HEALTH_STATE_UNSPECIFIED, err
+	}
+	hosts := resolver.AvailableMembers()
```
Comment (laniehei) [inline service/frontend/health_check.go:73]: return if no hosts to prevent panic later on

**Review by laniehei — APPROVED** [review]

File: service/frontend/admin_handler.go (hunk 2 above)
Comment (yux0) [reply service/frontend/admin_handler.go:259]: this is the interface definition. I want to have the request to have the room to pass in parameters.

File: service/frontend/health_check.go (hunk 3 above)
Comment (yux0) [reply service/frontend/health_check.go:82]: updated

File: service/frontend/health_check.go (hunk 4 above)
Comment (yux0) [reply service/frontend/health_check.go:73]: good catch. updated.

File: common/dynamicconfig/constants.go (hunk 1 above)
Comment (yux0) [reply common/dynamicconfig/constants.go]: updated.

**Review by yycptt — APPROVED** [review]

**Pull request merged**